### DISCUSSION TITLE
Whatsapp Business sync

### DISCUSSION THREAD

**Discussion by etiennea** [twentyhq/twenty#7296] [Ideas]
## Scope & Context

Pretty Self explanatory but could really add a lot of value and make this CRM stand out. 

## Current behavior

Whatsapp messages have to be added manually

**Comment by Bonapara** [comment]
Great idea!

**Comment by sekroots** [comment]
Awesome.  Anyone else  want it?

**Comment by mmaryo** [comment]
In some countries, WhatsApp is the first communication channel for leads and clients.
Using the Inbox to display and respond through WhatsApp and link the WhatsApp messages to the Contact and Company brings a lot of value!

**Comment by jlfguthrie** [comment]
Whatsapp integration should be considered fundamental.

**Comment by gb-123-git** [comment]
+1 for WhatsApp integration

**Comment by tek365** [comment]
Yes, after Gmail, Outlook, Pop/iMap, next is WhatsApp on my request list

**Comment by eduardohartz** [comment]
+1

A lot of businesses use WhatsApp to contact clients, more then email, it would be a great addition to be able to save messages within a Person.

**Comment by Lunorien** [comment]
Yes!

**Comment by moorenot** [comment]
Here in Brazil, we’re seeing a growing demand for this feature, especially among micro, small, and medium-sized businesses, as most of their sales come through this channel.

**Comment by jordan-chalupka** [comment]
Great idea! Would love to see this added

**Comment by ColectivoWeb** [comment]
+1

**Comment by westmuzo1** [comment]
Let's see how this goes.

**Comment by dntrply** [comment]
+1 whatsapp is ubiquitous in India

**Comment by zimin-iliya** [comment]
Yes please

**Comment by FreeSynergy** [comment]
Why not generally messengers? Telegram and X are also favorite messengers (Matrix is a free and decentral messenger)
If there is a possibiltity to abstract messenger apps and to extend %e class, than it is very easy to implement this. I don't like WhatsApp, but an implementation of the favorits is very good.

**Reply by Bonapara** [reply]
When we'll do it, we will abstract everything to ensure it's easy to add other messaging systems.

**Reply by Novapixel1010** [reply]
Being a more privacy focus person. YES for matrix or telegram.

**Reply by laforetcan** [reply]
We really need this for our company. Is there anywork started on this feature? We are thinking about implementing it ourselves.

**Reply by LaszloDev** [reply]
> When we'll do it, we will abstract everything to ensure it's easy to add other messaging systems.

Is there any work in process on supporting other messaging channels, which could be utilized by the upcoming apps?
I like to see how I can contribute to getting something implemented, and looking for what's the current state.

**Reply by 2bfree1** [reply]
Last I spoke to Devs in Discord, he said he never even used AI before.

Though it was January that's when we decided to just create python agentic workflows. Everyone with their own dashboard and seeing information they care about.  

Powerdialer of leads to call that day. No button pushing unless you wanted to create a new workflow

**Reply by fendle** [reply]
Hey, when you implement it, it would be good to have as well the possibility to get automatically the average response time by chat and how long the first answer took.

This helps sales manager to see if the team is answering in a proper time

**Comment by laforetcan** [comment]
Hi everyone!

I wanted to share that we're planning to start working on implementing WhatsApp integration next week for our own business needs. Given @Bonapara's comment about abstracting the messaging system to make it easy to add other platforms, we'll aim to build our implementation with that flexibility in mind.

We'll keep this discussion updated with our progress and share what we learn along the way. Hoping this can contribute to the broader 
Twenty ecosystem and help move this much-requested feature forward.

Looking forward to collaborating with the community on this! 🚀

**Reply by koz-ai** [reply]
Hi, when the integration for Whatsapp will be available? How can I contribute to it?

**Reply by 2bfree1** [reply]
Hi @tonoli 👋

Thanks for sharing your plans and for aiming to keep the WhatsApp integration flexible for other channels.

I’m very interested in this work because I’d like to add calling + SMS on top of Twenty in an upgrade-safe way (no long-term fork, ideally just using whatever messaging abstraction / hooks you end up with). I’ve recently been able to “vibe code” similar SMS + phone features into another app we use, so I’d love to align what I’m building with your approach instead of reinventing a separate messaging layer.

A few questions:

Do you already have a WIP repo/branch or design doc for the WhatsApp integration?

Are you planning this as a core PR to Twenty, or as something more like an extension/add-on that plugs into defined interfaces?

Is there a place (Discord/Slack/issue) where you’re discussing the messaging abstraction so others can contribute requirements (e.g., SMS/voice flows)?

Happy to share my calling/SMS use cases and help test or contribute wherever it’s useful.

**Reply by Bonapara** [reply]
Thanks!
Apps are coming soon, if you want to start working on the project now you should try using https://www.npmjs.com/package/twenty-cli (BETA)

**Reply by sapter** [reply]
I'd like to help out @tonoli @2bfree1. How can we collaborate?

**Comment by fendle** [comment]
Do you plan as well a kind of notification in Twenty CRM? Somehow you have to inform the user that someone answered a message.

**Reply by Bonapara** [reply]
Sure We plan to add an inbox to Twenty in the future but we are fully focused on extensibility at the moment!

**Reply by BOHEUS** [reply]
Notifications will be done in Q4 this year, maybe earlier depending how fast other things will be done (mainly real life sync, surfacing system objects and mail attachments) https://github.com/twentyhq/core-team-issues/issues/210

**Reply by fendle** [reply]
Than I would propose to you, that you implement WhatsApp business sync only after your inbox. Customers who are answering/writing to you expect quickly an answer and don't want to wait. E-Mail is a different thing, where people can wait.

We are using for example Chatwoot to interact with our customers and o sync the conversations via N8N to twenty CRM. 

It would be good, if you built a data model for WhatsApp messages, that I could reuse it for the messages from Chatwoot

**Reply by BOHEUS** [reply]
Integration should be done after mail attachments and notifications are done, more info here https://github.com/twentyhq/twenty/pull/17390#issuecomment-3794522204

It'll be done in stages either way as there's too much things to implement in one go

**Comment by slarrain** [comment]
Not only in India WhatsApp is big but also in all of Latin America. All sells are done using WhatsApp. So a CRM, nowadays, kind of needs a way of integrating because emails is just not enough.
I'm thinking of something similar to the email integration. Where you can read the messages between a user and a customer/prospects. It would also help when directing an LLM to Twenty, so it would have all context.

**Comment by victoMR** [comment]
+1 — Here in Mexico, WhatsApp is fundamental. I'm currently using this CRM (https://www.gohighlevel.com/henri-homepage-ads-page), but if I could help make Twenty work like that, it would be great.

**Comment by Facecloth4523** [comment]
I just got n8n pulling my whatsapp messages through evolution API and save to the CRM. But they get saved as notes which can be edited by anyone and everyone. Searched for a solution and laded here. I need this asap. Whatsapp is our main communication channel with client and partners

**Reply by ManasMadrecha** [reply]
Are you self hosting everything: twenty + evolution + n8n? 

You could store the messages in separate custom object (not notes), and make this object's certain fields read-only for others (timestamp, content).

**Reply by Facecloth4523** [reply]
Yes I am. I have only been able to insert the WhatsApp messages as notes.

If you could guide me to the right direction I would be super thankful

**Reply by ManasMadrecha** [reply]
Which self hosting vps are you using?

**Reply by Facecloth4523** [reply]
No vps. It's on our own hardware

**Reply by ManasMadrecha** [reply]
Regarding the custom object, you can create it like this https://docs.twenty.com/user-guide/data-model/capabilities/objects#creating-a-new-custom-object called Whatsapp Messages. 

and then in permissions in settings, set the field-level permissions of this custom object to only see-only (https://docs.twenty.com/user-guide/permissions-access/capabilities/permissions#field-permissions)

**Comment by MohamedMushraf-dev** [comment]
Yes whatsapp +1 for integration
